pkg/cmd/roachtest/tests/tpcc.go

```diff
@@ -1866,3 +1886,27 @@ func getTpccLabels(
 
 	return labels
 }
+
+// This function converts exporter.SnapshotTick to openmetrics into a buffer
+func getHistogramOpenmetrics(
+	exporter exporter.Exporter, snapshots map[string][]exporter.SnapshotTick,
+) (*bytes.Buffer, error) {
+	perfBuf := bytes.NewBuffer([]byte{})
+	writer := io.Writer(perfBuf)
+
+	exporter.Init(&writer)
+	for _, snaps := range snapshots {
+		for _, s := range snaps {
+			h := hdrhistogram.Import(s.Hist)
+			if err := exporter.SnapshotAndWrite(h, s.Now, s.Elapsed, &s.Name); err != nil {
+				return nil, errors.Wrapf(err, "failed to write snapshot for histogram %q", s.Name)
+			}
+		}
+	}
+
+	if err := exporter.Close(nil); err != nil {
```

Comment:
Please move this to defer right after Init.

Reply:
Actually, there's already an `exporter.Close()` in `CreateStatsFileInClusterFromExporterWithPrefix`, so I think I'll remove this already